harmony/harmony.py

```diff
@@ -173,7 +173,36 @@ def __repr__(self) -> str:
 _valid_shapefile_exts = ', '.join((_shapefile_exts_to_mimes.keys()))
 
 
-class Request:
+class BaseRequest:
+    """A Harmony base request with the CMR collection. It is the base class of all harmony requests.
+
+    Args:
+        collection: The CMR collection that should be queried
+
+    Returns:
+        A Harmony Request instance
+    """
+
+    def __init__(self,
+                 collection: Collection):
```

Comment:
Given what we want for the hackfest, it probably doesn't make sense for `collection` to be a property of the base class, or at least not a required parameter of the constructor.

Reply:
Makes sense. I will make it an optional keyword parameter.